src/Controller/Docs/FunctionsController.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace XdebugDotOrg\Controller\Docs;
+
+use XdebugDotOrg\Core\HtmlResponse;
+use XdebugDotOrg\Model\FunctionDescription;
+use XdebugDotOrg\Model\Functions;
+use XdebugDotOrg\Controller\DocsController;
+
+class FunctionsController
+{
+    /**
+     * @var array
+     */
+    private static $functions = null;
+
+    public function single(FunctionDescription $function) : HtmlResponse
+    {
+        return new HtmlResponse($function, 'docs/function.php');
+    }
+
+    /**
+     * @return array<string, FunctionDescription>
+     */
+    private static function getFunctions() : array
+    {
+        if (self::$functions !== null) {
+            return self::$functions;
+        }
+
+        $functions = [];
+
+        $function_include_dir = dirname( __DIR__, 3) . '/html/docs/include/functions/';
+
+        if (!file_exists($function_include_dir )) {
+            throw new \Exception($function_include_dir . ' should exist');
+        }
+
+        $f = glob( $function_include_dir . '*' );
+
+        foreach ( $f as $filename ) {
+            if ( !is_file( $filename ) ) {
+                continue;
+            }
+
+            $contents = file( $filename );
+            $function = trim( $contents[0], "= \n" );
+            $version = null;
+        
+            if (strpos( $function, ',' ) !== false) {
+                list( $function, $version ) = explode( ',', $function );
+            }
+
+            $arguments = trim( $contents[3] );
+
+            $functions[] = new FunctionDescription(
+                $function,
+                DocsController::add_links(trim( $contents[1] )),
+                self::do_format_data( array_slice( $contents, 6 ) ),
+                $version,
+                trim( $contents[2] ),
+                $arguments && $arguments !== 'none' ? $arguments : null,
+                constant( DocsController::class . '::' . trim( $contents[4] ) )
+            );
+        }
+
+        return $functions;
+    }
+
+    /**
+     * @return Setting[]
+     */
+    public static function getRelatedFunctions(int $func) : array
+    {
+        $functions = array_filter(
+            self::getFunctions(),
+            function ($function) use ($func) {
+                return $func & $function->type;
+            }
+        );
+
+        ksort($functions);
+
+        return $functions;
+    }
+
+
+    /**
+     * @return Setting[]
+     */
+    public static function all() : HtmlResponse
+    {
+        $functions = self::getFunctions();
+
+        ksort($functions);
+
+        return new HtmlResponse(new Functions($functions), 'docs/all_functions.php');
+    }
+
+    private static function do_format_data_TXT(array $data)
```

Comment:
I only migrated the HTML that was in that settings away. Functions are still handled by this parsing step (that can remain, because my changes shouldn't touch it).

Reply:
Ah, right. I think I would like to change that to just normal HTML as well, but there is no rush for that.